File: moviejava/app/src/main/java/com/example/moviejava/data/remote/genres/GenresService.java
package com.example.moviejava.data.remote.genres;


import com.example.moviejava.model.JsonObject;
import com.example.moviejava.model.genres.FromJson;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GenresService {
    @GET("genre/movie/list")
    Observable<FromJson> getListGenres();

    @GET("discover/movie")
    Observable<JsonObject> getListFilmByGenres(@Query("without_genres") int id, @Query("page") int page);
}

File: movie/app/src/main/java/com/example/movie/data/remote/genres/GenresService.kt
package com.example.movie.data.remote.genres

import com.example.movie.data.remote.Constant
import com.example.movie.model.genres.FromJson
import com.example.movie.model.JsonObject
import retrofit2.http.GET
import retrofit2.http.Query

interface GenresService {
    @GET("genre/movie/list")
    suspend fun getListGenres(): FromJson

    @GET("discover/movie")
    suspend fun getListFilmByGenres(@Query("without_genres") id: Int,@Query("page") page:Int): JsonObject
}